6.4.2 cherrypick deprecation (#728)
ROCm/rocPRIM: CHANGELOG.md

Full documentation for rocPRIM is available at [https://rocm.docs.amd.com/projects/rocPRIM/en/latest/](https://rocm.docs.amd.com/projects/rocPRIM/en/latest/).

## rocPRIM 3.4.1 for ROCm 6.4.2

### Upcoming changes

> * The next major release may change the template parameters of warp and block algorithms.

Reviewer: I would reword this a bit.

```suggestion
* Changes to the template parameters of warp and block algorithms will be made in an upcoming release.
```

Reviewer 2: Done.
